lib/logstash/inputs/http_poller.rb

```diff
@@ -42,6 +54,20 @@ def register
 
     @logger.info("Registering http_poller Input", :type => @type, :schedule => @schedule, :timeout => @timeout)
 
+    @url_field = ecs_select[disabled: "[#{metadata_target}][request][url]", v1: "[url][full]"]
+    @method_field = ecs_select[disabled: "[#{metadata_target}][request][method]", v1: "[http][request][method]"]
+    @host_field = ecs_select[disabled: "[#{metadata_target}][host]", v1: "[host][hostname]"]
+    @response_code_field = ecs_select[disabled: "[#{metadata_target}][code]", v1: "[http][response][status][code]"]
+    @response_headers_field = ecs_select[disabled: "[#{metadata_target}][response_headers]", v1: "[#{metadata_target}][input][http_poller][response][headers]"]
+    @response_message_field = ecs_select[disabled: "[#{metadata_target}][response_message]", v1: "[#{metadata_target}][input][http_poller][response][status][message]"]
+    @response_time_field = ecs_select[disabled: "[#{metadata_target}][runtime_seconds]", v1: "[#{metadata_target}][input][http_poller][response][time][second]"]
+    @request_retried_field = ecs_select[disabled: "[#{metadata_target}][times_retried]", v1: "[#{metadata_target}][input][http_poller][request][retried]"]
+    @request_name_field = ecs_select[disabled: "[#{metadata_target}][name]", v1: "[#{metadata_target}][input][http_poller][request][name]"]
+    @request_manticore_field = ecs_select[disabled: "[#{metadata_target}][request]", v1: "[#{metadata_target}][input][http_poller][request][original]"]
+    @error_msg_field = ecs_select[disabled: "[http_request_failure][error]", v1: "[error][message]"]
+    @stack_trace_field = ecs_select[disabled: "[http_request_failure][backtrace]", v1: "[error][stack_trace]"]
```

Comment:
since these are LS/Ruby specific I am a bit hesitant having these present by default, it's easy to drop but still
a [code comment suggest this used to live in meta-data](https://github.com/logstash-plugins/logstash-input-http_poller/blob/v5.0.2/lib/logstash/inputs/http_poller.rb#L205-L206)
@yaauie any preference on moving Ruby exception details under `@metadata` rather than using `error.*` ?

Reply:
🤔 

In a way, we have two possible types of events that this plugin emits:
 - events retrieved from an external HTTP service
 - events indicating failure to retrieve from the external HTTP service.

In that sense, these error fields are not metadata -- they are _only_ used on the later type of event and are in fact the primary data about those events -- use of the `event.*`, `error.*`, etc. are useful and appropriate.

If a pipeline author needs to be able to differentiate between these error events and events that were retrieved, and can do so with a conditional (such as: `if "_http_request_failure" in [tags] { ... }`).

I would prefer to settle on a consistent naming pattern in our code to make this a little more obvious. Perhaps we name all ivars for metadata-about-http-request-that-resulted-in-events with `metadata_*_field`, and the ivars for `failed-request-data` fields with `failure_*_field`?

Reply:
I am convinced they should be two types of events
for events retrieved from HTTP poller, metadata route to [$target_metadata][input][http_poller]
for failure event, data map to `error.*`, `url.*`, `host.*`, `http.*`

If user follows our suggestion to config `target` when ECS compatibility is enabled, even the event contains `host.hostname` or `url.full`, they will not be mapped to ECS fields. There is no guarantee how data shaped in events. Should we try to map the target fields to ECS fields, `target.host.hostname` to `host.hostname`?

Reply:
> If user follows our suggestion to config `target` when ECS compatibility is enabled, even the event contains `host.hostname` or `url.full`, they will not be mapped to ECS fields. There is no guarantee how data shaped in events. Should we try to map the target fields to ECS fields, `target.host.hostname` to `host.hostname`? 

the user understands the `target` is there for the retrieved (http) payload so I would leave the potential un-targeting to them (to be handled by the pipeline logic). so I think we're good here. 
just a reminder I thing the `target` will only be used as a last resort when the codec does not provide one.